cirq-core/cirq/ops/classically_controlled_operation.py

```diff
@@ -148,39 +163,77 @@ def _circuit_diagram_info_(
     def _json_dict_(self) -> Dict[str, Any]:
         return {
             'cirq_type': self.__class__.__name__,
-            'conditions': self._control_keys,
+            'conditions': self._conditions,
             'sub_operation': self._sub_operation,
         }
 
     def _act_on_(self, args: 'cirq.ActOnArgs') -> bool:
-        def not_zero(measurement):
-            return any(i != 0 for i in measurement)
-
-        measurements = [
-            args.log_of_measurement_results.get(str(key), str(key)) for key in self._control_keys
-        ]
-        missing = [m for m in measurements if isinstance(m, str)]
-        if missing:
-            raise ValueError(f'Measurement keys {missing} missing when performing {self}')
-        if all(not_zero(measurement) for measurement in measurements):
-            protocols.act_on(self._sub_operation, args)
+        for condition in self._conditions:
+            keys, expr = condition.keys, condition.expr
+            missing = [str(k) for k in keys if str(k) not in args.log_of_measurement_results]
+            if missing:
+                raise ValueError(f'Measurement keys {missing} missing when performing {self}')
+            replacements = {
+                f'x{i}': args.log_of_measurement_results[str(k)][0] for i, k in enumerate(keys)
```

Comment:
Only getting zeroth index of the measurement. Need to convert the measurement from a bitstring to an integer. This means `args` will have to also keep a record `log_of_measurement_bases` in order to support qudits.